### ISSUE TITLE
Basic Pre-aggregator Design and Implementation

### ISSUE THREAD

**Issue by zhanghaohit** [4paradigm/OpenMLDB#1202]
**Is your feature request related to a problem? Please describe.**
sub-task of long-time window feature

**Describe the solution you'd like**
- Upon `Put` request, we check if there are associated pre-aggregators.
- If Yes, we update these pre-aggregators

Pre-aggregator update steps:
- update its internal states
- if the current mini-window is full/complete, close the current mini-window and insert the new pre-aggregated mini-window result into the pre-aggregation table
- if the new tuple is out of the boundary of the current mini-window, re-read the corresponding entry, update and put back

*As first attempt, we only consider one op: sum*

**Additional context**
long-time window feature

This issue depends on Issue #1197 and #1217

For this issue, we only consider the normal data insert.
For the scenario where there are already some data in the table when issuing deploy, we will consider in another issue #1204

**Comment by zhanghaohit** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [long time window - compute engine](https://trello.com/c/XMMF5TXl/6-long-time-window-compute-engine)
